src/DynamoManipulation/DynamoManipulationExtension.cs

```diff
@@ -118,18 +121,28 @@ public string Name
 
         public void Dispose()
         {
-            manipulatorDaemon.KillAll();
+            manipulatorDaemon?.KillAll();
 
             UnregisterEventHandlers();
         }
 
         public void Startup(ViewStartupParams viewStartupParams)
         {
+            if (!viewStartupParams.IsGeometryLibraryLoaded)
+            {
+                MessageLogged(LogMessage.Info($"GeometryLibrary was not loaded, skipping {nameof(DynamoManipulationExtension)}"));
+                return;
+            };
             manipulatorDaemon = ManipulatorDaemon.Create(new NodeManipulatorFactoryLoader());
         }
 
         public void Loaded(ViewLoadedParams viewLoadedParams)
         {
+            if (!viewLoadedParams.StartupParams.IsGeometryLibraryLoaded)
```

Comment:
Do we need the `IsGeometryLibraryLoaded` check in both `Startup `and `Loaded `functions?

Reply:
yes, I'd like to avoid partially initializing as much as possible.